modules/common/src/encoding.ts

```diff
@@ -133,6 +134,23 @@ export function decodeProposalId(
   };
 }
 
+/** Gets the extended version of a proposal id from the compact one */
+export const getExtendedProposalId = (proposalId: string): string => {
+  if (!proposalId.match(/^(0x[A-Fa-f0-9]{40})_(0x[A-Fa-f0-9]{1,64})$/)) {
+    throw new Error("Invalid proposalId");
+  }
+  const splits = proposalId.split("_");
+  return splits[0].toLowerCase() + "_" + hexZeroPad(splits[1], 32);
+};
+/** Gets the compact version of a proposal id from the extended one */
+export const getCompactProposalId = (proposalId: string): string => {
+  if (!proposalId.match(/^(0x[A-Fa-f0-9]{40})_(0x[A-Fa-f0-9]{1,64})$/)) {
+    throw new Error("Invalid proposalId");
+  }
+  const splits = proposalId.split("_");
+  return splits[0].toLowerCase() + "_0x" + parseInt(splits[1]).toString(16);
```

Comment:
Does `parseInt()` accept hex strings in a standardized way?

Reply:
Yes, at least thaths what [mozilla mdn](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/parseInt) and [w3schools](https://www.w3schools.com/jsref/jsref_parseint.asp) says